File: flutter_frontend/extract.dart
import 'dart:io';

void main() {
  final RegExp exp = RegExp(r'AppTheme\.(\w+)');
  final Set<String> colors = {};
  
  final dir = Directory('lib');
  for (final entity in dir.listSync(recursive: true)) {
    if (entity is File && entity.path.endsWith('.dart')) {
      final content = entity.readAsStringSync();
      final matches = exp.allMatches(content);
      for (final m in matches) {
        colors.add(m.group(1)!);
      }
    }
  }
  print(colors.join('\n'));
}

File: flutter_frontend/extract.py
import os, re
colors = set()
for root, _, files in os.walk('lib'):
    for file in files:
        if file.endswith('.dart'):
            with open(os.path.join(root, file), 'r', encoding='utf-8') as f:
                content = f.read()
                matches = re.findall(r'AppTheme\.(\w+)', content)
                colors.update(matches)
print(", ".join(colors))
